src/mvPlotting.cpp

```diff
@@ -458,40 +392,34 @@ DearPyGui::draw_plot(ImDrawList* drawlist, mvAppItem& item, mvPlotConfig& config
 		if (config._useColorMap)
 			ImPlot::PopColormap();
 
-		config._queried = ImPlot::IsPlotQueried();
-
-		if (config._queried)
-		{
-			ImPlotLimits area = ImPlot::GetPlotQuery();
-			config._queryArea[0] = area.X.Min;
-			config._queryArea[1] = area.X.Max;
-			config._queryArea[2] = area.Y.Min;
-			config._queryArea[3] = area.Y.Max;
-		}
-
-		if (item.config.callback != nullptr && config._queried)
+		if (item.config.callback != nullptr && config.rects.size() > 0)
```

Comment:
It turns out that `config.rects.size() > 0` is not an adequate replacement for `config._queried`.

The old version only invoked the callback when the query rect changed. The new version, however, calls it _every frame_ even if there are no changes to rect coordinates. This puts more burden on the callback as it probably needs to check whether the query rects have changed since the last call.

We can add a "query dirty" flag and set it whenever the user adds or deletes a drag rect, or when `ImPlot::DragRect` returns `true` (meaning the rect has changed). Then only schedule the callback if the dirty bit is on. This also means that callback invocation should go after `DragRect` below.